### PR TITLE
Migrate bicep example: 101/cosmosdb-private-endpoint

### PR DESCRIPTION
**Pull request by StephenWeatherford**
Added bicep support to this sample by integrating bicep.main from the example in https://github.com/Azure/bicep/tree/main/docs/examples/101/cosmosdb-private-endpoint

https://github.com/Azure/bicep/pull/3148

### CODE REVIEW

**Comment by StephenWeatherford** [comment]
FYI @ThomasWeiss, @MickaëlMottet

File: quickstarts/microsoft.documentdb/cosmosdb-private-endpoint/main.bicep [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+@description('Cosmos DB account name (must be lower-case)')
+@maxLength(44)
+param accountName string
+
+@allowed([
+  'Enabled'
+  'Disabled'
+])
+@description('Enable public network traffic to access the account; if set to Disabled, public network traffic will be blocked even before the private endpoint is created')
```
Comment (bmoore-msft) [inline quickstarts/microsoft.documentdb/cosmosdb-private-endpoint/main.bicep:15]: move description to the top

File: quickstarts/microsoft.documentdb/cosmosdb-private-endpoint/main.bicep [hunk 2]
```diff
@@ -0,0 +1,85 @@
+@description('Location for all resources.')
+param location string = resourceGroup().location
+
+@description('Virtual network name')
+param virtualNetworkName string
+
+@description('Cosmos DB account name (must be lower-case)')
+@maxLength(44)
```
Comment (bmoore-msft) [inline quickstarts/microsoft.documentdb/cosmosdb-private-endpoint/main.bicep:8]: docs say this is 3-31 chars

File: quickstarts/microsoft.documentdb/cosmosdb-private-endpoint/main.bicep [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  properties: {
+    addressSpace: {
+      addressPrefixes: [
+        '172.20.0.0/16'
+      ]
+    }
+  }
+}
+
+resource subNet 'Microsoft.Network/virtualNetworks/subnets@2020-06-01' = {
```
Comment (bmoore-msft) [inline quickstarts/microsoft.documentdb/cosmosdb-private-endpoint/main.bicep:41]: remove camel case in ```subnet```

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

File: quickstarts/microsoft.documentdb/cosmosdb-private-endpoint/main.bicep (hunk 2 above)
Comment (StephenWeatherford) [reply quickstarts/microsoft.documentdb/cosmosdb-private-endpoint/main.bicep:8]: Portal:
image

Comment (StephenWeatherford) [reply quickstarts/microsoft.documentdb/cosmosdb-private-endpoint/main.bicep:8]: fixed to match portql

**Comment by bmoore-msft** [comment]
@StephenWeatherford - can you check the merge conflicts?

**Pull request merged**